bin/testEnvLocalK8S.sh

```diff
@@ -128,49 +133,82 @@ function startLocalApiserver() {
     # make sure apiserver is actually alive
     kill -0 $(cat $LOG_DIR/apiserver.pid)
 
-    echo "Started local etcd and apiserver !"
+    # Really need to make sure that API Server is up before proceed further
+    waitForApiServer
+    printf "Started local etcd and apiserver!\n"
+}
+
+function ensureLocalApiServer() {
+    kubectl get nodes 2>/dev/null || startLocalApiserver
+}
+
+function createIstioConfigmap() {
```

Comment:
In general - we're trying to save the output in $OUT_DIR, not tmp.

Maybe we can do a bit more - and apply the entire istio setup ? (of course no pod will be started).

Reply:
Since you guys are mounting config files into pods, so it will not be possible in local environment.  I see two possibility, 1 - refactor the way a process gets configuration and instead of mounting it inside of a container, directly get config information (client-go api) from a configmap and unmarshal it into in memory structs, 2 - extract config into files and place them at location the process expect to see it.  It is doable if the location of config for a process is customizable, not hard coded like /etc/pilot/config for example.

Reply:
Yes, location should not be hardcoded - we can add some env variable to control it (PILOT_CONFIG ?). 

We may also do (1) - and watch so we handle config changes, but (2) should be done either way, we also support 
cases where there is no k8s.